pkg/manifest/mutator.go

```
@@ -0,0 +1,313 @@
+		mergedArgs := make([]string, len(existingArgs))
+		copy(mergedArgs, existingArgs)
+
+		for _, newArg := range newArgs {
+			key := extractArgKey(newArg)
+			if idx, exists := argIndex[key]; exists {
+				// Replace existing arg
+				mergedArgs[idx] = newArg
+			} else {
+				// Append new arg
+				argIndex[key] = len(mergedArgs)
+				mergedArgs = append(mergedArgs, newArg)
```

golangci-lint (2.11.4)

[error] append to slice `mergedArgs` with non-zero initialized length

(makezero)